src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -505,11 +512,6 @@ internal void WriteCentralDirectoryFileHeader()
                 extraFieldLength = (ushort)bigExtraFieldLength;
             }
 
-            if (_hasUnicodeEntryNameOrComment)
-                _generalPurposeBitFlag |= BitFlagValues.UnicodeFileNameAndComment;
-            else
-                _generalPurposeBitFlag &= ~BitFlagValues.UnicodeFileNameAndComment;
-
```

Comment:
I'm ok with removing this if we are directly modifying the general purpose bit flag when detecting the comment or filename encoding.